creator-node/src/snapbackSM.js

```diff
@@ -348,60 +366,94 @@ class SnapbackSM {
+   */
+  async monitorSecondarySync (userWallet, primaryClockValue, secondaryClockValue, secondaryUrl) {
+    const startTime = Date.now()
+
+    // Define axios request object for secondary sync status request
+    const syncMonitoringRequestParameters = {
       method: 'get',
       baseURL: secondaryUrl,
-      url: `/sync_status/${syncWallet}`,
+      url: `/sync_status/${userWallet}`,
       responseType: 'json'
     }
+
+    // TODO syncAttemptCompleted does nothing - refactor this
     let syncAttemptCompleted = false
+    let secondaryClockValAfterSync = null
     while (!syncAttemptCompleted) {
       try {
-        let syncMonitoringResp = await axios(syncMonitoringRequestParameters)
-        let respData = syncMonitoringResp.data.data
-        this.log(`processSync ${syncWallet} secondary response: ${JSON.stringify(respData)}`)
+        const syncMonitoringResp = await axios(syncMonitoringRequestParameters)
+
+        const respData = syncMonitoringResp.data.data
+        this.log(`processSync ${userWallet} secondary response: ${JSON.stringify(respData)}`)
+
         // A success response does not necessarily guarantee completion, validate response data to confirm
         // Returned secondaryClockValue can be greater than the cached primaryClockValue if a client write was initiated
         //    after primaryClockValue cached and resulting sync is monitored
         if (respData.clockValue >= primaryClockValue) {
+          secondaryClockValAfterSync = respData.clockValue
+
+          this.log(`Sync for ${userWallet} at ${secondaryUrl} from ${secondaryClockValue} to ${secondaryClockValAfterSync} for primaryClockVal ${primaryClockValue} completed in ${Date.now() - startTime}ms`)
           syncAttemptCompleted = true
-          this.log(`processSync ${syncWallet} clockValue from secondary:${respData.clockValue}, primary:${primaryClockValue}`)
           break
         }
       } catch (e) {
-        this.log(`processSync ${syncWallet} error querying sync_status: ${e}`)
+        this.log(`processSync ${userWallet} error querying sync_status: ${e}`)
       }
+
+      // Stop retrying if max sync monitoring duration exceeded
       if (Date.now() - startTime > MaxSyncMonitoringDurationInMs) {
-        this.log(`ERROR: processSync ${syncWallet} timeout for ${syncWallet}`)
+        this.log(`ERROR: processSync ${userWallet} timed out`)
         syncAttemptCompleted = true
+        break
       }
+
       // Delay between retries
       await utils.timeout(SyncMonitoringRetryDelay)
     }
-    let duration = Date.now() - startTime
-    if (!syncAttemptCompleted) {
-      this.log(`Sync for ${syncWallet} at ${secondaryUrl} completed in ${duration}ms`)
+
+    // enqueue another sync if secondary is still behind
+    // TODO max retry limit
+    if (!secondaryClockValAfterSync || secondaryClockValAfterSync < primaryClockValue) {
+      await this.enqueueSecondarySync({
+        userWallet,
```

Comment:
is there a maximum number of times we should retry this before giving up forever? even with low pri enqueuing this still may end up unnecessarily occupying the queue

Reply:
yeah there does need to be a retry limit, lets discuss tm

Reply:
discussed offline, this is sufficently unlikely to occur and will handle separately